Add action sidebar on web layout

vladtf opened the pull request:
Changes:
* Add action bar on right side
* Restrict max width of body
* Add maximum width to login form
* Set persistence on web to SESSION

Commit:
add maximum width to login form

AndreeaAdascalitei committed:
new contributer: capsunica (#213)

* new contributer: capsunica

* new contributer: capsunica

dependabot[bot] committed:
Bump addressable from 2.7.0 to 2.8.0 in /android (#214)

Bumps [addressable](https://github.com/sporkmonger/addressable) from 2.7.0 to 2.8.0.
- [Release notes](https://github.com/sporkmonger/addressable/releases)
- [Changelog](https://github.com/sporkmonger/addressable/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sporkmonger/addressable/compare/addressable-2.7.0...addressable-2.8.0)

---
updated-dependencies:
- dependency-name: addressable
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

IoanaAlexandru committed:
Fix workflow failing on code review. (#223)

IoanaAlexandru committed:
Improve automated review (#224)

* Update dangerfile (take #1)

* Fix dangerfile.

* Test with code changes.

* Test with unrelated pubspec change.

* Update dangerfile again.

* Test with actual version change.

* Make danger check for changelogs.

* Add test changelog.

* Final (?) dangerfile change.

* Revert test changes.

* Move remote_config import back

IoanaAlexandru committed:
Prevent use of S.of(context) via automated review. (#225)

* Prevent use of S.of(context).

* Add test code.

* Fix dangerfile.

* Check only dart files.

* Revert "Add test code."

This reverts commit bd975da2

* Move remote_config import back

AlexandraPavel committed:
Updated README.md, added a new contributor (#216)

GhiaraD committed:
Add Badea Dragos to contributors list (#226)

alinpahontu2912 committed:
Add Alin Pahontu to Contributors List (#221)

* Added myself to contributers

* Added myself to contributers

* Added myself to contributers

* Added myself to contributers

* Added myself to contributers

* Added myself to contributers(+ Ș/ț)

Co-authored-by: stafy2912 <stafy2912@gmail.com>

StefanP-EQ committed:
Stefan Popa - Update Contributors

stefanp0pa committed:
Update README.md

StefanP-EQ committed:
Revert pubspec.lock

StefanP-EQ committed:
Merge branch 'andreew/update_contributors' of github.com:AndreiPopa21/acs-upb-mobile into andreew/update_contributors

stefanp0pa committed:
Update pubspec.lock

stefanp0pa committed:
Update pubspec.lock

IoanaAlexandru committed:
Sort people by last name. #trivial (#228)

* Sort people by last name.

* Bump build number.

Co-authored-by: Andrei-Constantin Mirciu <38398944+andreicmirciu@users.noreply.github.com>

AndreiMirica19 committed:
Add Andrei Mirica to contributors (#212)

* update_contributors

* Update pubspec.lock

* Update README.md

stefanp0pa committed:
Merge branch 'master' into andreew/update_contributors

stefanp0pa committed:
Merge pull request #229 from AndreiPopa21/andreew/update_contributors

Stefan Popa - Update Contributors

IoanaAlexandru committed:
Allow aii.pub.ro unsafe connections. #trivial (#227)

* Allow unsafe traffic from aii.pub.ro.

* Only allow HTTP fo aii.pub.ro.

* Fix some whitespace.

* Bump version.

* Allow HTTP on iOS as well.

* Indent Info.plist using tabs.

* Missed a tab.

IoanaAlexandru committed:
UI improvements (#178)

* Remove bottom blue line in tab bar.

* Change SelectableFormField to use Chips.

* Improve Chip theme.

* Make dividers more consistent.

Couldn't find the exact setting but this looks decent.

* Add custom theme for dark mode.

* First attempt at fixing relevance picker.

* Fix logic for relevance picker.

* Select "Anyone" when no custom node is selected.

* Call relevancePicker.onChanged whenever it changes.

* Disable public filter options if user doesn't have permission.

* Fix ChoiceChip theme.

* Make RelevancePicker a FormField everywhere.

* Reuse ChipFormField in RelevanceFormField.

* Move filterIconColor to resources/theme.dart.

* Remove some duplicate code.

* Remove S.of(context).

* Fix default relevance in picker.

* Use chips on filter page.

This also fixes #5.

* Use chips in FAQ page.

* Fix tests.

* Make flutter version in workflows more specific.

Workflows now fail on GH but work locally, so I think it's because the
1.24.x version somehow gets matched to 2.3.0-24.1.pre-beta instead of
1.24.0-10.2.pre.

* Revert workflow changes.

No longer seems to be necessary and the previous version didn't work well anyway.

* Prep for release.

* Bump version and update changelog.

* Add comma for formatting.

Co-authored-by: Andrei-Constantin Mirciu <38398944+andreicmirciu@users.noreply.github.com>

IoanaAlexandru committed:
Bump patch version for release.

IoanaAlexandru committed:
Update gradle to fix failing release.

https://stackoverflow.com/questions/68255521/could-not-get-https-google-bintray-com-maven-metadata-xml-received-stat

IoanaAlexandru committed:
Fix GH release failing.

IoanaAlexandru committed:
Fix GH release failing (take #2).

IoanaAlexandru committed:
Fix some typos in CONTRIBUTING.md.

IoanaAlexandru committed:
Update documentation about GitHub Actions. (#232)

* Update documentation about GitHub Actions.

* Remove extra backtick.

* Remove duplicate "the".

* Add more info about tags.

* Add prompt to not ask the bot for review.

vladtf committed:
Add action bar on right side

IoanaAlexandru committed:
Make missing pubspec.yaml version update a failure instead of a warning.

IoanaAlexandru committed:
Check that there's a space between // or /// and the actual comment (#236)

* Check space after comment slashes.

* Add test case for linter.

* s/elif/elsif

Ruby iz hard

* Add EOF test case.

* Fix regex for regular comments.

* Make the example build.

* Check for spaces before comment, and more than 3 slashes.

* Add more test cases.

* Fix syntax error.

* Check for block comments.

* Add test case for block comments.

* Remove check for space before comment as that's handled by the formatter.

* Update comment.

* Remove test cases.

IoanaAlexandru committed:
Fix comment linter complaining about URLs.

AndreiMirica19 committed:
If the phone is in dark mode,the splash screen will be in dark mode. (#238)

* If the phone is in dark mode,the splash screen will be in dark mode.

* Update pubspec.yaml

Co-authored-by: Ioana Alexandru <ioanaa.alexandru98@gmail.com>

AndreiMirica19 committed:
Image is cropped if it doesn't fit. (#239)

* Image is cropped if it doesn't fit.
Smartphone:
Device:One Plus 9 Pro
Os:Android 11

* Image is cropped if it doesn't fit.
Smartphone:
Device:One Plus 9 Pro
Os:Android 11

* Image is cropped if it doesn't fit.
Smartphone:
Device:One Plus 9 Pro
Os:Android 11

* Image is cropped if it doesn't fit.
Smartphone:
Device:One Plus 9 Pro
Os:Android 11

AndreeaAdascalitei committed:
Uploaded the website icon in add/edit mode. (#233)

* Uploaded the website icon in add/edit mode. Also, added the necessary label for text (labelUploadWebsiteIcon) and moved convertToPNG to Utils class

* added ',' to intl_en and intl_ro

* changed the line 220 with the correct form of formIconColor

* deleted the empty lines and used the full path of user.dart(line 4)

* changed in intl_en.arb

* changed WebsiteIcon (from website_view)

* changed the name to oldImage, used the if short and ??

* initialized oldImage ()\otherwise error in tests

* changed the initialization of imageWidget

* Use an actual text field instead of imitating it.

It has text input behaviour disabled.

* Show file name in image picker field.

* Remove commented code.

* Add a clear button for the image field.

* changed labelUploadWebsiteIcon to labelWebsiteIcon

* space after // and explained what image in WebsiteIcon stands for

* explained what image in WebsiteIcon stands for and incremented build number in pubspec

* added iconPath in Website and used it accordingly

* created a path for public and private icon websites

* delete website icon from storage when website is deleted

* changed the string for labelWebsiteIcon, both in ro and en, and changed the attributes and methos in Statet from private in public

* changed the requirement of iconPath

* deleted the converttoPNG from edit_profile_page.dart

* updated the converttoPNG method

* changed the variable iconPath to a getter

* formatting

Co-authored-by: Ioana Alexandru <ioanaa.alexandru98@gmail.com>

AndreiMirica19 committed:
Toast overlapped bug (#246)

* Deleted the edit profile succesfully toast.

* Deleted the edit profile succesfully toast.

IoanaAlexandru committed:
Deploy security rules from repo (#248)

* Set up firebase CLI with firestore indexes & rules.

* Add firestore deployment step.

* Update functions.

* Set up storage rules.

* Update documentation.

* Use "bash"  for command blocks instead of "cli".

* Use bash syntax highlighting for the other command block in the file as well.

* Replace "as close to possible" with "as close as possible".

* There's actually no command to fetch the current rules.

vladtf committed:
Add action_sidebar.dart
Restrict max width of body

vladtf committed:
Set persistence on web to SESSION

IoanaAlexandru commented on lib/pages/people/view/people_page.dart:
```diff
@@ -42,6 +42,7 @@ class _PeoplePageState extends State<PeoplePage> {
           onPressed: () {
             setState(() => searchClosed = !searchClosed);
           },
+          tooltip: 'Search'
```
All strings should be localized. See the internationalization section in the documentation:
https://github.com/student-hub/acs-upb-mobile/blob/master/CONTRIBUTING.md#internationalization

vladtf replied:
fc8b2b706a6833e30a7bb8c7ab7cafd24abce559

AndreiMirica19 committed:
Updated merging rules. (#251)

IoanaAlexandru commented on lib/widgets/action_sidebar.dart:
```diff
@@ -0,0 +1,219 @@
+
+class _ActionSideBarState extends State<ActionSideBar>
+    with SingleTickerProviderStateMixin {
+  _ActionSideBarState({this.isExtended = false});
```
What's the point of the parameter in this constructor? It isn't actually used anywhere.

If you just want to define the default value, just do `bool isExtended = false` below.

vladtf replied:
Yes, it's not really used so could be moved to field:

https://github.com/student-hub/acs-upb-mobile/pull/250/commits/8c0e90bd1db807c70ea5422527345d766711da4c#diff-91af2d9745e86244940959ec39c99cca9b8b43c1502042d85f16fcc6a311030bR17

IoanaAlexandru commented on lib/widgets/action_sidebar.dart:
```diff
@@ -0,0 +1,219 @@
+
+  bool isExtended;
+
+  AnimationController _controller;
```
FWIW, this class is private, so there isn't really much point in having private properties imo.

vladtf replied:
Rename to controller:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/8c0e90bd1db807c70ea5422527345d766711da4c#diff-91af2d9745e86244940959ec39c99cca9b8b43c1502042d85f16fcc6a311030bR17

IoanaAlexandru commented on lib/widgets/action_sidebar.dart:
```diff
@@ -0,0 +1,219 @@
+
+  @override
+  Widget build(BuildContext context) {
+    if (MediaQuery.of(context).size.width > 1100) {
```
Maybe put this value in a const, you're using it in multiple places and risk forgetting to change all of them if you ever wanna change it.

vladtf replied:
Extracted as a const value:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/8c0e90bd1db807c70ea5422527345d766711da4c#diff-91af2d9745e86244940959ec39c99cca9b8b43c1502042d85f16fcc6a311030bR20

IoanaAlexandru commented on lib/widgets/action_sidebar.dart:
```diff
@@ -0,0 +1,219 @@
+
+class ActionBarButton extends StatelessWidget {
+  const ActionBarButton(this.onTap, this.controller,
+      {this.maxWidth = 50, this.maxHeight = 50});
```
Again defining parameters that are never used? Or am I missing something?

vladtf replied:
Removed values from constructor:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/8c0e90bd1db807c70ea5422527345d766711da4c#diff-91af2d9745e86244940959ec39c99cca9b8b43c1502042d85f16fcc6a311030bR135-R137

IoanaAlexandru commented on lib/widgets/action_sidebar.dart:
```diff
@@ -0,0 +1,219 @@
+  }
+}
+
+class BezierPoint {
```
All these classes that aren't needed outside of this file should be private imo.

vladtf replied:
Made private:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/31536835b8d109bd3ef7b078038b0eafbd696c57#diff-91af2d9745e86244940959ec39c99cca9b8b43c1502042d85f16fcc6a311030bR179-R180

IoanaAlexandru commented on lib/widgets/scaffold.dart:
```diff
@@ -163,4 +167,58 @@ class AppScaffold extends StatelessWidget {
     );
   }
+
+  ///
```
There should be no empty lines in dartdoc style:
https://dart.dev/guides/language/effective-dart/documentation

vladtf replied:
Removed empty lines:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/31536835b8d109bd3ef7b078038b0eafbd696c57#diff-91af2d9745e86244940959ec39c99cca9b8b43c1502042d85f16fcc6a311030bR179-R180

IoanaAlexandru commented on lib/widgets/scaffold.dart:
```diff
@@ -163,4 +167,58 @@ class AppScaffold extends StatelessWidget {
   }
+
+  ///
+  /// Build content body based on is web version.
+  /// When web version ad an action bar to right side.
```
This doesn't really make sense. Maybe "Build content body based on platform. On web, there are action buttons on the right side."?

vladtf replied:
Changed comments:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/8c0e90bd1db807c70ea5422527345d766711da4c#diff-91af2d9745e86244940959ec39c99cca9b8b43c1502042d85f16fcc6a311030bR135-R137

IoanaAlexandru commented on lib/widgets/scaffold.dart:
```diff
@@ -163,4 +167,58 @@ class AppScaffold extends StatelessWidget {
+  /// When web version ad an action bar to right side.
+  ///
+  Widget buildBody(BuildContext context, {bool enableContent = false}) {
+    final List<Widget> actionButtons =
+        !kIsWeb // check to avoid building action button twice in mobile
+            ? List.empty()
+            : ([leading] + actions)
+                .map((action) => buildMaterialActionButton(action, context,
+                    enableContent: enableContent))
+                .where((element) => element != null)
+                .toList();
+
+    return kIsWeb && actionButtons.isNotEmpty
+        ? Stack(
+            children: [
+              Center(
+                  child: ConstrainedBox(
+                      constraints: BoxConstraints(maxWidth: maxBodyWidth),
+                      child: body)),
+              ActionSideBar(actionButtons),
+            ],
+          )
+        : body;
```
The variable and multiple conditions make this a bit hard to read. How about this:
```suggestion
    return !kIsWeb
        ? body
        : Stack(
            children: [
              Center(
                  child: ConstrainedBox(
                      constraints: BoxConstraints(maxWidth: maxBodyWidth),
                      child: body)),
              ActionSideBar(
                  ([leading] + actions)
                    .map((action) => buildMaterialActionButton(action, context,
                        enableContent: enableContent))
                    .where((element) => element != null)
                    .toList(),
              ),
            ],
          );
```

vladtf replied:
Still want to check if there is action to be added to sidebar, otherwise sidebar shouldn't be added.

vladtf replied:
Moved mapping through actions:
https://github.com/student-hub/acs-upb-mobile/pull/250/files#diff-2804fa2a5ae6726cac5690433203db74702f6ee38877cb902a9d5095c419b578R175-R176
https://github.com/student-hub/acs-upb-mobile/pull/250/files#diff-2804fa2a5ae6726cac5690433203db74702f6ee38877cb902a9d5095c419b578R175-R176

IoanaAlexandru replied:
I would argue that even if there are no actions to put in the sidebar, the maxWidth constraint should still be applied on web.

So I still think you should use my proposed code structure, and inside the sidebar check if the list of actions is empty; if it is, simply don't build anything.

vladtf replied:
Right, missed the part that if action list is empty max width constraint won't be applied , thanks.

vladtf replied:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/dad30a96ea179228206a08abb53d168a6ef85933

IoanaAlexandru commented on lib/widgets/action_sidebar.dart:
```diff
@@ -0,0 +1,219 @@
+import 'package:flutter/cupertino.dart';
```
I wonder if it would make more sense to move this (and maybe the scaffold?) into the navigation directory. WDYT?

vladtf replied:
I'm not sure if it do really belongs to navigation since it just replaces app bar and must be on same level as scaffold. About moving scaffold, I think another idea is to group widgets that are responsible for layout.

vladtf replied:
Changes:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/31536835b8d109bd3ef7b078038b0eafbd696c57#diff-91af2d9745e86244940959ec39c99cca9b8b43c1502042d85f16fcc6a311030bR179-R180
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/31536835b8d109bd3ef7b078038b0eafbd696c57#diff-91af2d9745e86244940959ec39c99cca9b8b43c1502042d85f16fcc6a311030bR179-R180

IoanaAlexandru replied:
Well, the bottom navigation bar *is* in the navigation directory, and the side bar is  similar I guess.
It's a bit hard to define what should go into "layout" I think, as essentially everything is a layout...

vladtf replied:
I agree that scaffold must be placed in same directory as bottom navigation bar but not sure if navigation folder is the best place because there is already  routes.dart and as I do remember @RazvanRotaru  already added some classes that manages navigation.

vladtf replied:
Navigation changes:
https://github.com/student-hub/acs-upb-mobile/tree/web/layout/navigation/lib/navigation

RazvanRotaru replied:
I agree with Ioana, this widget is welcomed in /navigation/view

IoanaAlexandru commented on lib/widgets/scaffold.dart:
```diff
@@ -163,4 +167,58 @@ class AppScaffold extends StatelessWidget {
+  ///
+  /// Wrap simple action button to enable hover and splash effects
+  ///
+  AspectRatio buildMaterialActionButton(
```
This feels like it should be the sidebar's job. Why not make the sidebar constructor take a list of AppScaffoldAction as a parameter instead?

vladtf replied:
Losing access to private function _widgetFromAction but could pass widgets from actions and inside sidebar class to wrap into material.

vladtf replied:
Moved to action sidebar class:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/8c0e90bd1db807c70ea5422527345d766711da4c#diff-2804fa2a5ae6726cac5690433203db74702f6ee38877cb902a9d5095c419b578R187-R191

IoanaAlexandru replied:
What if we move _widgetFromAction to AppScaffoldAction and make it a public method, like `toWidget({bool enabled, BuildContext context})`?

So you could just call the sidebar like:
```dart
ActionSideBar(actions: [leading] + actions, enabled: enableContent);
```

vladtf replied:
Yes, definitely don't belongs to  AppScaffold.

vladtf replied:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/49d688c89623ace6d8c3826d72aaaec2f17da1aa

IoanaAlexandru commented on web/index.html:
```diff
@@ -54,6 +54,7 @@
   // Initialize Firebase
   firebase.initializeApp(firebaseConfig);
   firebase.analytics();
+  firebase.auth().setPersistence(firebase.auth.Auth.Persistence.SESSION);
```
Maybe add a small comment to say why this is necessary.

vladtf replied:
Add comment:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/8c0e90bd1db807c70ea5422527345d766711da4c#diff-2804fa2a5ae6726cac5690433203db74702f6ee38877cb902a9d5095c419b578R187-R191

IoanaAlexandru replied:
I said why it's necessary, I can tell from the code that it's enabling "auth persistence", I just don't get what it is and why we need it.

vladtf replied:
Here is the discussion about this issue:
https://github.com/student-hub/acs-upb-mobile/issues/207

IoanaAlexandru replied:
Maybe add a link to the issue in the comment?

vladtf replied:
In other projects that are managed from Jira I saw using ticket names as reference, but considering that GitHub uses issue number I'll add a full link to related issue.

vladtf committed:
Add internationalization for search action

vladtf committed:
Move buildMaterialActionButton to action_sidebar.dart
Add comment about persistence in index.html
Move constructor named paramteres to field values

vladtf committed:
Make _BezierPoint class private
Remove empty dartdoc lines

vladtf committed:
Use S.current.actionSearch

vladtf committed:
Merge branch 'master' into web/layout/sidebar/pr

# Conflicts:
#	lib/pages/people/view/people_page.dart

vladtf committed:
Add maxWidth constraint when on web even if sidebar is empty

vladtf committed:
Move _widgetFromAction and buildMaterialActionButton to AppScaffoldAction class

vladtf committed:
Fix warning messages

vladtf committed:
Merge branch 'web/layout/main' into web/layout/sidebar/pr

# Conflicts:
#	.github/linter/Dangerfile
#	CONTRIBUTING.md
#	lib/pages/timetable/view/events/add_event_view.dart

vladtf committed:
Import app_scaffold_action.dart

vladtf committed:
Add issue link as reference

vladtf committed:
Use "0 as double" instead of "0.toDouble()"

vladtf commented on lib/widgets/action_sidebar.dart:
```diff
@@ -0,0 +1,221 @@
+                alignment: Alignment.centerRight,
+                color: Theme.of(context).bottomAppBarColor,
+                child: RotationTransition(
+                    turns: Tween(begin: 0.5, end: 0.0).animate(controller),
```
Workaround:  0 as double

vladtf committed:
Format code

IoanaAlexandru commented on lib/authentication/view/login_view.dart:
```diff
@@ -141,114 +141,122 @@ class _LoginViewState extends State<LoginView> {
-                constraints: BoxConstraints(
-                  maxHeight: MediaQuery.of(context).size.height / 4,
-                  maxWidth: MediaQuery.of(context).size.width,
+        body: Center(
```
What do changes in this file have to do with this PR? Please make a separate PR if you wanna improve the login page layout.

RazvanRotaru commented on lib/pages/home/home_page.dart:
```diff
@@ -33,7 +34,7 @@ class HomePage extends StatelessWidget {
       body: ListView(
         children: [
-          if (authProvider.isAuthenticated) ProfileCard(),
+          if (authProvider.isAuthenticated) const ProfileCard(),
```
I believe that adding `const`, in this case, will result in an undesired behavior.
As I experienced it, if a anonymous user is authenticated than the `ProfileCard` won't show up.

vladtf committed:
Revert login pages changes

vladtf committed:
Move AppScaffoldAction batck to scaffold.dart

vladtf committed:
Fix code format

vladtf committed:
Revert const

vladtf committed:
Move navigation widgets to navigation/view
